contracts/test/semantics.ts

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: Apache-2.0
+
+import { ethers } from 'hardhat';
+import { expect } from 'chai';
+import { SemanticTests } from '../typechain-types/contracts/tests/SemanticTests';
+import { SemanticTests__factory } from '../typechain-types/factories/contracts/tests';
+
+const ERROR_NUM =
+  '0x1023456789abcdef1023456789abcdef1023456789abcdef1023456789abcdef';
+
+describe('EVM Semantics', () => {
+  let c: SemanticTests;
+  let chainId: number;
+
+  before(async () => {
+    const f = (await ethers.getContractFactory(
+      'SemanticTests',
+    )) as SemanticTests__factory;
+    c = await f.deploy();
+    await c.deployed();
+    chainId = (await c.provider.getNetwork()).chainId;
+  });
+
+  it('eth_call maximum return length vs gas limit', async () => {
+    if (chainId != 31337) {
```

Comment:
I would expect sapphire-contracts tests not to run on hardhat node. On the CI, do we test it against hardhat node too?